File: Client/Android/app/src/main/java/com/example/examinapp/dataaccess/dtos/base/BaseResponse.java
package com.example.examinapp.dataaccess.dtos.base;

import java.util.Date;

public class BaseResponse {

    private boolean isSuccess;
    private int status;
    private String message;
    private Date respondDateTime;

    public BaseResponse(boolean isSuccess, int status, String message, Date respondDateTime) {
        this.isSuccess = isSuccess;
        this.status = status;
        this.message = message;
        this.respondDateTime = respondDateTime;
    }

    public boolean getIsSuccess() {
        return isSuccess;
    }

    public void setSuccess(boolean success) {
        isSuccess = success;
    }

    public int getStatus() {
        return status;
    }

    public void setStatus(int status) {
        this.status = status;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public Date getRespondDateTime() {
        return respondDateTime;
    }

    public void setRespondDateTime(Date respondDateTime) {
        this.respondDateTime = respondDateTime;
    }
}

File: Server/CTSE-Exam-In-App/DTOs/Base/BaseResponse.cs
using System;
using System.Net;

namespace CTSE_Exam_In_App.DTOs.Base
{
    public class BaseResponse
    {
        public bool IsSuccess { get; set; }

        public int Status => IsSuccess ? (int)HttpStatusCode.OK : (int)HttpStatusCode.BadRequest;

        public string Message { get; set; }
        public DateTime RespondDateTime { get; set; }
        
        public BaseResponse()
        {
            RespondDateTime = DateTime.Now;
        }
    }
}
